overlord/state/task.go

```diff
@@ -33,13 +33,14 @@ type progress struct {
 //
 // See Change for more details.
 type Task struct {
-	state    *State
-	id       string
-	kind     string
-	summary  string
-	status   Status
-	progress progress
-	data     customData
+	state      *State
+	id         string
+	kind       string
+	summary    string
+	status     Status
+	progress   progress
+	data       customData
+	waitingFor []*Task
```

Comment:
Nitpick: can we please name these as `waitTasks` ("wait-tasks" for JSON)?  The term "waitingFor" tastes a little bit like this list should be empty once this task is ready to go, which won't be the case. It's arguably a very minor point, but it won't hurt to rename either.